.travis.yml

```diff
@@ -1,100 +1,73 @@
-    - env: CABALVER=1.24 GHCVER=8.0.1 DEPLOY_GITHUB_RELEASE=true
-      compiler: ": #GHC 8.0.1"
-      addons: {apt: {packages: [cabal-install-1.24,ghc-8.0.1], sources: [hvr-ghc]}}
-    - env: CABALVER=1.24 GHCVER=8.0.1 DEPLOY_GITHUB_RELEASE=true
-      compiler: ": #GHC 8.0.1"
-      os: osx
+  # Build only on macOS
+  - env: BUILD=stack
+    compiler: ": #stack default osx"
+    os: osx
+
+# Build only master and release tags
+branches:
+  only:
+  - master
+  - /^\d+\.\d+\.\d+(\.\d+)?$/
+
 
 before_install:
- - unset CC
- - export PATH=/opt/ghc/$GHCVER/bin:/opt/cabal/$CABALVER/bin:$PATH
- # Work around travis issue by updating Homebrew: https://github.com/travis-ci/travis-ci/issues/8552
- - if [ "$TRAVIS_OS_NAME" = osx ];
-   then
-     brew update;
-     brew install cabal-install;
-     brew install gnu-sed --with-default-names;
-   fi
+  # Using compiler above sets CC to an invalid value, so unset it
+  - unset CC
+  - CABALARGS=""
+  - export PATH=/opt/ghc/$GHCVER/bin:/opt/cabal/$CABALVER/bin:$HOME/.local/bin:/opt/alex/$ALEXVER/bin:/opt/happy/$HAPPYVER/bin:$HOME/.cabal/bin:$PATH
+  - | # Install stack
+    mkdir -p ~/.local/bin
+    travis_retry curl --insecure -L https://get.haskellstack.org/stable/osx-x86_64.tar.gz | tar xz --strip-components=1 --include '*/stack' -C ~/.local/bin
+    # Use the more reliable S3 mirror of Hackage
+    mkdir -p $HOME/.cabal
+    echo 'remote-repo: hackage.haskell.org:http://hackage.fpcomplete.com/' > $HOME/.cabal/config
+    echo 'remote-repo-cache: $HOME/.cabal/packages' >> $HOME/.cabal/config
 
-install:
- - cabal --version
- - echo "$(ghc --version) [$(ghc --print-project-git-commit-id 2> /dev/null || echo '?')]"
- - if [ -f $HOME/.cabal/packages/hackage.haskell.org/00-index.tar.gz ];
-   then
-     zcat $HOME/.cabal/packages/hackage.haskell.org/00-index.tar.gz >
-          $HOME/.cabal/packages/hackage.haskell.org/00-index.tar;
-   fi
- - travis_retry cabal update -v
- - sed -i 's/^jobs:/-- jobs:/' ${HOME}/.cabal/config
- - cabal install --only-dependencies --enable-tests --enable-benchmarks --dry -v > installplan.txt
- - sed -i -e '1,/^Resolving /d' installplan.txt; cat installplan.txt
 
-# check whether current requested install-plan matches cached package-db snapshot
- - if diff -u $HOME/.cabsnap/installplan.txt installplan.txt;
-   then
-     echo "cabal build-cache HIT";
-     rm -rfv .ghc;
-     cp -a $HOME/.cabsnap/ghc $HOME/.ghc;
-     cp -a $HOME/.cabsnap/lib $HOME/.cabsnap/share $HOME/.cabsnap/bin $HOME/.cabal/;
-   else
-     echo "cabal build-cache MISS";
-     rm -rf $HOME/.cabsnap;
-     mkdir -p $HOME/.ghc $HOME/.cabal/lib $HOME/.cabal/share $HOME/.cabal/bin;
-     cabal install --only-dependencies --enable-tests --enable-benchmarks;
-   fi
+install:
+  - echo "$(ghc --version) [$(ghc --print-project-git-commit-id 2> /dev/null || echo '?')]"
+  - if [ -f configure.ac ]; then autoreconf -i; fi
+  - set -ex
+  - stack --no-terminal --install-ghc $ARGS test --bench --only-dependencies
```

Comment:
What is `$ARGS` set to?

Also, I don't think we need to run tests or benchmarks.  We already do that in the Nix build.  We only need the Travis build for generating OS X binaries

Reply:
Ideally it should be enough but would not be possible to have specific errors or perf problems in OS X?

Reply:
Note that this is building dependencies only, we're not running the tests of Dhall itself

Reply:
Then shouldn't this dependency-only command also only use `build` instead of `test --bench`?

My understanding is that the actual build command for the `dhall` packages is just `stack build` (i.e. doesn't run tests or benchmarks), meaning that we don't need to install test-related dependencies or benchmark-related dependencies.

Reply:
Yep, I changed it to `build` and removed the unneeded flags